Modules/TRD/src/TrackletCountCheck.cxx

```diff
@@ -0,0 +1,190 @@
+  ILOG(Debug, Support) << "using Upperthreshold Per Timeframe= " << mThresholdMeanHighPerTimeFrame << ENDM;
+
+  mThresholdMeanLowPerTimeFrame = getFromConfig<float>(mCustomParameters, "LowerthresholdPerTimeFrame", 600.f);
+  ILOG(Debug, Support) << "using Lowerthreshold Per Timeframe= " << mThresholdMeanLowPerTimeFrame << ENDM;
+
+  mThresholdMeanHighPerTrigger = getFromConfig<float>(mCustomParameters, "UpperthresholdPerTrigger", 520.f);
+  ILOG(Debug, Support) << "using Upperthreshold Per Trigger= " << mThresholdMeanHighPerTrigger << ENDM;
+
+  mThresholdMeanLowPerTrigger = getFromConfig<float>(mCustomParameters, "LowerthresholdPerTrigger", 500.f);
+  ILOG(Debug, Support) << "using Lowerthreshold Per Trigger= " << mThresholdMeanLowPerTrigger << ENDM;
+
+  mStatThresholdPerTrigger = getFromConfig<int>(mCustomParameters, "StatThresholdPerTrigger", 1000);
+  ILOG(Debug, Support) << "using StatThreshold Per Trigger= " << mStatThresholdPerTrigger << ENDM;
+}
+
+Quality TrackletCountCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  mResultPertrigger = Quality::Null;
+  mResultPerTimeFrame = Quality::Null;
+
+  for (auto& [moName, mo] : *moMap) {
+
+    (void)moName;
+    if (mo->getName() == "trackletsperevent") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+      if (h == nullptr) {
+        // ILOG(Debug, Support) << "Requested Histogram type does not match with the Histogram in source" << ENDM;
+        continue;
+      }
+      // warning about statistics available
+      TPaveText* msg1 = new TPaveText(0.3, 0.7, 0.7, 0.9, "NDC"); // check option "br","NDC"
+      h->GetListOfFunctions()->Add(msg1);
+      msg1->SetTextSize(10);
+      int Entries = h->GetEntries();
+      if (Entries > mStatThresholdPerTrigger) {
+        msg1->AddText(TString::Format("Hist Can't be ignored. Stat is enough. Entries: %d > Threshold: %d", Entries, mStatThresholdPerTrigger));
+        // msg1->SetTextColor(kGreen);
+      } else if (Entries > 0) {
+        msg1->AddText(TString::Format("Hist Can be ignored. Stat is low. Entries: %d < Threshold: %d", Entries, mStatThresholdPerTrigger));
+        // msg1->SetTextColor(kYellow);
+      } else if (Entries == 0) {
+        msg1->AddText(TString::Format("Hist is empty. Entries: %d < Threshold: %d", Entries, mStatThresholdPerTrigger));
+        msg1->SetTextColor(kRed);
+      }
+
+      // Warning about triggers without any tracklets
+      int UnderFlowTrackletsPerTrigger = h->GetBinContent(0);
+      if (UnderFlowTrackletsPerTrigger > 0.) {
+        msg1->AddText(TString::Format("Triggers without Tracklets: %d", UnderFlowTrackletsPerTrigger));
+      }
+
+      // applying check
+      float MeanTrackletPertrigger = h->GetMean();
+      if (MeanTrackletPertrigger > mThresholdMeanLowPerTrigger && MeanTrackletPertrigger < mThresholdMeanHighPerTrigger) {
+        TText* Checkmsg = msg1->AddText("Mean Per Trigger is found in bound region: ok");
```

Comment:
maybe add to the text the mean (it its not anyhow visible from the statistics box of the histogram) and the boundaries?